public/components/filter_bar/filter_editor.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCodeEditor,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiFormRow,
+  EuiPopoverTitle,
+  EuiSpacer,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { i18n } from '@osd/i18n';
+
+interface Props {
+  content: string;
+  label: string;
+  onSubmit: (content: string, label: string) => void;
+  onCancel: () => void;
+}
+
+const isFilterValid = (content: string) => {
+  try {
+    return Boolean(JSON.parse(content));
+  } catch (e) {
+    return false;
+  }
+};
+
+export const FilterEditor = ({ content, label, onSubmit, onCancel }: Props) => {
+  const [filterLabel, setFilterLabel] = useState<string>(label);
+  const [filterContent, setFilterContent] = useState<string>(content);
+
+  const renderEditor = () => {
+    return (
+      <EuiFormRow
+        label={i18n.translate('maps.filter.filterEditor.parameters', {
+          defaultMessage: 'Spatial filter parameters',
+        })}
+        fullWidth={true}
+      >
+        <EuiCodeEditor
+          value={filterContent}
+          onChange={setFilterContent}
```

Comment:
How about use callback here?
```
  const handleFilterContentChange = useCallback((value: string) => {
    setFilterContent(value);
  }, []);
```

Reply:
i don't think there is a difference since it only updates the value in useState.